src/service/iot_config.proto

```diff
@@ -365,6 +365,53 @@ message admin_remove_key_req_v1 {
 
 message admin_key_res_v1 {}
 
+message gateway_info {
+  /// The asserted h3 (string) location of the gateway.
+  // Empty string if the gateway is not found
+  string location = 1;
+  bytes address = 2;
+  bytes owner = 3;
+  bool is_full_hotspot = 4;
+  /// the transmit gain value of the gateway in dbi x 10
+  // For example 1 dbi = 10, 15 dbi = 150
+  int32 gain = 5;
+  region region = 6;
+  blockchain_region_params_v1 region_params = 7;
+}
+
+enum gateway_error_reason {
+  reason_none = 0;
+  gateway_not_found = 1;
+}
+
+message gateway_error_res {
+  bytes address = 1;
+  gateway_error_reason error = 2;
+}
+
+/// Look up the details of a given hotspot public key
+message gateway_info_req_v1 {
+  /// The pubkey_bin address of the gateway to look up
+  bytes address = 1;
+}
+
+message gateway_info_res_v1 {
+  /// Timestamp of response
+  uint64 timestamp = 1;
+  oneof result {
+    gateway_info info = 2;
+    gateway_error_res error = 3;
```

Comment:
given the above suggestion about using tonic/grpc status responses to convey errors i would pull the `gateway_info` back up to the top level here outside of any `oneof` and again depending on our network topography this will need a signature to verify by the oracle as well

Reply:
addressed here: e8f3a601a1ff0713430ee9e1b70406f3efeb39ff